example/src/stories/Button.test.ts

```diff
@@ -0,0 +1,107 @@
+import { render, screen } from "@testing-library/vue";
+import * as stories from "./Button.stories";
+import Button from "./Button.vue";
+import { test, vi } from "vitest";
+import { composeStories, composeStory, setProjectAnnotations } from "../../../dist";
+import { expectTypeOf } from 'expect-type';
+import { Meta } from '@storybook/vue3';
+
+// example with composeStories, returns an object with all stories composed with args/decorators
+const { CSF3Primary } = composeStories(stories);
+
+// example with composeStory, returns a single story composed with args/decorators
+const Secondary = composeStory(stories.CSF2Secondary, stories.default);
+
+test('renders primary button', () => {
+  render(CSF3Primary({ label: 'Hello world' }));
```

Comment:
Hey @yannbf, I would like to clarify a few points here:

In Use-case 1, if we are testing against CFS Annotations using @testing-library/vue, everything looks good. 👍

In Use-case 2, if we want to test against the storybook renderer, we should use @storybook/testing-vue3 based on the implementation of @storybook/vue3 renderer.

At the moment, it seems we are in USE CASE 1: Passing the tests doesn't necessarily mean that @storybook/vue renders the story correctly. However, it validates the preview API and CSF annotation, and proves that delegating rendering to @testing-library/vue with the provided annotations works perfectly.

Actually, what we should do is test the @storybook/vue3 renderer again, so we need to expose the render function.

Reply:
Hey @chakAs3 ! Thank you so much for checking this out. Although I agree that we should use the render function form @storybook/vue3, this package will likely be deprecated soon, in favor of moving this utility directly inside of the @storybook/vue3 package, just like it happened with @storybook/testing-react which moved to @storybook/react. Therefore I don't think we should expose the render function, if the only intention is to use it in this package

Reply:
Ok cool, I agreed that this should move to monorepo, 😄 I was about to suggest this. for me, testing is a core feature of storybook, 
therefore I will just copy the code for now, and this fails 1 test due to the old issue I tried to explain many times,
 **__"render function should return representation of the DOM and not another function_** , 
this creates also miss behavior inside @storybook/vue3 renderer.  

I think we should ship this package for now even though is failing,  ( cause this. is the purpose of it to test both csf and @storybook/vue3 

what do you suggest @yannbf ?

Reply:
Alright I figured it all out! Managed to get this working correctly by using all code from Storybook core libraries and the @storybook/vue3 annotations as well. I'll make a release and at some point start working on bringing this to the @storybook/vue3 package in the monorepo, so users can do `import { composeStories } from '@storybook/vue3'` and migrate away from this library.

Thank you so much @chakAs3, this helped me think this all through, especially the representation of the DOM part, which was crucial for the tests. <3